src/core_ocean/shared/mpas_ocn_gm.F

```diff
@@ -450,14 +401,34 @@ subroutine ocn_GM_compute_Bolus_velocity(statePool, diagnosticsPool, &
             ! mrp Performance Note: Replace with a logical array on init later
             if (cell1==iCell) then
                 iCellSelf = 1
+                sVal = cell1
             elseif (cell2==iCell) then
                 iCellSelf = 2
+                sVal = cell2
             else
                 print *, 'iCellSelf, something went wrong',cell1,cell2,iCell,nCells
             endif
             dcEdgeInv = 1.0_RKIND / dcEdge(iEdge)
             areaEdge = dcEdge(iEdge) * dvEdge(iEdge)
 
+            ! prep dz, dTdz and dSdz for this column
+            do k = 2, maxLevelCell(sVal)
+              dzTop(k) = 0.5_RKIND * ( layerThickness(k-1,sVal) + layerThickness(k,sVal) )
+              dTdzTop(k) = (  activeTracers(indexTemperature,k-1,sVal) &
+                            - activeTracers(indexTemperature,k  ,sVal) ) &
+                          / dzTop(k)
+              dSdzTop(k) = (  activeTracers(indexSalinity,k-1,sVal) &
+                            - activeTracers(indexSalinity,k  ,sVal) ) &
+                          / dzTop(k)
+            end do
+            ! mrp performance: could move above cell loop later
+            dzTop(1) = -1e-15_RKIND
+            dTdzTop(1) = -1e-15_RKIND
+            dSdzTop(1) = -1e-15_RKIND
+            dzTop(maxLevelCell(sVal)+1) = -1e-15_RKIND
+            dTdzTop(maxLevelCell(sVal)+1) = -1e-15_RKIND
+            dSdzTop(maxLevelCell(sVal)+1) = -1e-15_RKIND
+
```

Comment:
I think we need to compute dtdz and dsdz each time.  There are times when the stratification is computed for iCell but this is not consistent with iCellSelf.  This makes most sense to me comparing to the picture in Griffies et al 1998